templates/masterclasses.html

```diff
@@ -247,93 +245,100 @@ <h3 class="p-muted-heading u-sv2">Something missing?</h3>
 </script>
 
 <script defer>
-  let images = document.querySelectorAll('.p-card__image');
-  images.forEach(function (image) {
-    image.onerror = function () {
-      image.src = 'https://assets.ubuntu.com/v1/a647fb72-Copy%20of%20Canonical%20Slides%20Template.jpg';
-    }
-  });
+  document.addEventListener("DOMContentLoaded", () => {
+    const images = document.querySelectorAll('.p-card__image');
+    const resetSearch = document.getElementById('reset-search');
+    const searchInput = document.getElementById('search-input');
+    const submitButton = document.getElementById('submit-search');
+    const cards = document.querySelectorAll('.p-card');
+    const chips = document.querySelectorAll(".p-chip");
+
+    const setImageFallback = (image) => {
+      image.onerror = () => {
+        image.src = 'https://assets.ubuntu.com/v1/a647fb72-Copy%20of%20Canonical%20Slides%20Template.jpg';
+      };
+    };
+
+    const changeTag = (node, tag) => {
+      const clone = document.createElement(tag);
+      Array.from(node.attributes).forEach(({ name, value }) => clone.setAttribute(name, value));
+      while (node.firstChild) clone.appendChild(node.firstChild);
+      node.replaceWith(clone);
+      return clone;
+    };
+
+    const resetChips = () => {
+      document.querySelectorAll(".p-chip--positive").forEach((chip) => {
```

Comment:
This `querySelectorAll` logic hints at a possibility to select more than one tag at a time for filtering, which would be a great user experience, but instead, this is currently not supported. Could you please change the logic below (e.g. by using `searchParams.append` instead of `searchParams.set`, and then `searchParams.getAll` instead of `searchParams.get`) so that we can allow multi-tag filtering?

Reply:
Following our discussion, I've fixed the usage of querySelectorAll and search reset.

Reply:
[Ticket](https://warthogs.atlassian.net/browse/WD-14154?atlOrigin=eyJpIjoiMDU3NjM0ODZhMzA2NDUyZTgzOTc2MmMwOTEwNWE3NmYiLCJwIjoiaiJ9) for multi-tag filtering